front-end/src/shared/Recommendations.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import styled from 'styled-components';
+import { fetchRecommendedUsers } from '../api/RecommendationsApi';
+import { getUserIdFromJWT } from '../helpers/auth';
+import LoadingSpinner from '../shared/LoadingSpinner';
+import RecommendationsListItem from './RecommendationsListItem';
+
+class Recommendations extends React.Component {
+  state = {
+    recommendedUsers: [],
+    isLoading: true,
+    hasError: false,
+  }
+
+  componentDidMount() {
+    this.getRecommendedUsers();
+  }
+
+  getRecommendedUsers = async () => {
+    const userId = getUserIdFromJWT();
+
+    try {
+      const recommendedUsers = await fetchRecommendedUsers(userId);
+
+      this.setState({ recommendedUsers, isLoading: false });
+    } catch (err) {
+      this.setState({ isLoading: false, hasError: true });
+    }
+  }
+
+  handleInviteRequest = (userId) => async () => {
+    // TODO: add connection
+    alert(`TODO: Send invite to user ${userId}`);
+  }
+
+  handleIgnoreUser = (userId) => async () => {
+    // TODO: ignore user
+    alert(`TODO: Ignore user ${userId}`);
+  }
+
+  renderRecommendedUsersList = () => {
+    if (this.state.hasError) {
+      return <RecNotification>An error has occurred. Please try again.</RecNotification>
+    }
+
+    if (!this.state.recommendedUsers.length) {
+      return <RecNotification>No users to recommend!</RecNotification>
+    }
+
+    return (
+      <RecList>
+        {this.state.recommendedUsers.map((user) => {
```

Comment:
Kind of a silly question, but I can't seem to figure out where "user" comes from here

Reply:
`map` is a function that iterates over `recommendedUsers` and allows you to pass in a callback to apply some transformation to each element. Jon decided to call that variable `user`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

In the docs, `x` is the same as `user`

Reply:
Ahhh, I see now. 

Thanks for the explanation!!